### PR TITLE
selenium testing

### PR DESCRIPTION
**Pull request by fisx**
- script for running the docker selenium executable.
- trivial test case (everything else can come later).

### CODE REVIEW

**Comment by fisx** [comment]
- currently depends on branch `aula-docker-0.2`, but i want to merge that branch and revert f7f1b8c before landing this PR.

### issues

quay.io seems to have a hard time building today.  haven't investigated yet.

building the docker image locally might work (i only tried by hand), but running the test suite fails.  i have tried firefox and chromium with similar error messages (grid cannot connect to driver).  the problem definitely lies on the selenium side (the browser cannot be started), but other than that i have mostly walked through dead ends.

**Comment by fisx** [comment]
quay.io has recovered.

**Comment by fisx** [comment]
the issue was that chrome doesn't like to run with sandboxes inside docker (something about linux containers inside containers).  didn't resolve the firefox issue; chrome works fine now.

**Comment by fisx** [comment]
to merge the two branches in aula-docker and here, proceed as follows:

1. merge this branch.
1. revert the branch pinning commit on aula-docker-0.2.
1. merge aula-docker-0.2.
1. revert the branch pinning commit on this branch.
1. merge this branch again.

(4, 5 can be done locally and pushed to github when done.)

**Comment by fisx** [comment]
(updated last comment.)

**Comment by fisx** [comment]
See #907 for a more developed version of the above process.

File: .travis.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ before_install:
 
 script:
     - docker run --rm -it -v `pwd`:$AULA_SOURCE $AULA_IMAGE /bin/sh -c "$AULA_SOURCE/.travis/docker-hlint.sh"
-    - docker run --rm -it -v `pwd`:$AULA_SOURCE $AULA_IMAGE /bin/sh -c "$AULA_SOURCE/.travis/docker-build.sh --qc-max-success 1000 --qc-max-size 150"
+    - docker run --rm -it -v `pwd`:$AULA_SOURCE $AULA_IMAGE /bin/sh -c "$AULA_SOURCE/.travis/docker-build.sh --qc-max-success 100 --qc-max-size 100"
```
Comment (andorp) [inline .travis.yml]: We should keep the size parameter at 150. In some cases errors in paths test are not found using the default max-size 100.

File: tests/SeleniumSpec.hs [hunk 2]
```diff
@@ -3,6 +3,16 @@
...
 
+-- | selenium tests.
+--
+-- run these by running `make selenium` in a terminal in the aula-docker image.
+--
+-- for debugging, you have two options:
+--
+-- 1. sprinkle @getSource >>= writeFile "/page.html"@, @saveScreenshot "/screenshot.png"@
+--    over your 'WD' monads.
+-- 2. run find out the IP address of your docker image (run `ifconfig` in a terminal), and
```
Comment (andorp) [inline tests/SeleniumSpec.hs]: +1

**Comment by andorp** [comment]
So far so good.

**Pull request merged**